src/ameba/severity.cr

```diff
@@ -0,0 +1,56 @@
+module Ameba
+  enum Severity
+    Error
+    Warning
+    Refactoring
+
+    # Returns a symbol uniquely indicating severity.
+    #
+    # ```
+    # Severity::Warning.symbol # => 'W'
+    # ```
+    def symbol
+      to_s[0]
+    end
+
+    # Creates Severity by the name.
+    #
+    # ```
+    # Severity.from_name('refactoring') # => Severity::Refactoring
+    # Severity.from_name('foo-bar')     # => Exception: Incorrect severity name..
+    # ```
+    #
+    def self.from_name(name : String)
```

Comment:
There's already `Enum.parse` which does exactly this, why duplicate it?

Reply:
Nice, will change it. Thanks

Reply:
I meant using `Severity.parse` instead of this whole method ;)

Reply:
It doesn’t completely fit the needs ;)

Reply:
In what way it doesn't? :)

Reply:
@Sija please checkout the code and specs it explains better.

We want to report to the user the appropriate error and the options to fix his issue:

https://github.com/veelenga/ameba/blob/master/src/ameba/severity.cr#L26

```
$ ameba --fail-level WrongName
Error: Incorrect severity name WrongName. Try one of [Error, Warning, Refactoring]
```

The low level `Enum.parse` raises an `ArgumentError` and reports implementation details rather than the way how to fix the issue: `Unknown enum Severity value: WrongName (ArgumentError)`

Reply:
I thought so but wasn't sure that was your main motivation for the change. Wouldn't be cleaner then to overwrite the `.parse` method? If you'd still like to keep it this way I'd recommend to call `parse?` instead of rescuing the exception.

Reply:
Good point, will do